scilpy/tracking/tracker.py

```diff
@@ -253,38 +253,31 @@ def _get_line_both_directions(self, pos):
         #  This is a convenience, legacy function.
         #  The best practice is to not reseed a BitGenerator, rather to
         #  recreate a new one. This method is here for legacy reasons.
-        np.random.seed(np.uint32(hash((pos, self.rng_seed))))
-        line = [pos]
-
-        # Initialize returns true if initial directions at pos are valid.
-        if self.propagator.initialize(pos, self.track_forward_only):
-            # Forward
-            forward = self._propagate_line(True)
-            if len(forward) > 0:
-                forward.pop(0)
-                line.extend(forward)
-
-            # Backward
-            if not self.track_forward_only:
-                # Depending on the propagator, adding possibility to reset some
-                # parameters at this point.
-                self.propagator.start_backward()
-
-                backward = self._propagate_line(False)
-                if len(backward) > 0:
-                    line.reverse()
-                    line.pop()
-                    line.extend(backward)
-
-            # Clean streamline
-            if self.min_nbr_pts <= len(line) <= self.max_nbr_pts:
-                return line
+        np.random.seed(np.uint32(hash((seeding_pos, self.rng_seed))))
+
+        # Forward
+        line = [seeding_pos]
+        tracking_info = self.propagator.prepare_forward(seeding_pos)
+        if tracking_info == 'err':
```

Comment:
Could we place the `'err'` string in a global variable or an `Enum`? I think an enum would work well. It could be added to the `propagator` module. Something like :
```
class PropagationStatus(Enum):
    ERROR=1

# ...

if tracking_info == PropagationStatus.ERROR:
    # do stuff
```